### ISSUE TITLE
ArgoCD always shows login prompt when first accessed, even if current browser session has already authenticated with same OIDC identity provider

### ISSUE THREAD

**Issue by bbhuston** [argoproj/argo-cd#2535]
**Steps to reproduce bug:**
- Set up OIDC SSO with ArgoCD and your IdP
- Open new incognito browser session in Chrome
- Navigate to one of your apps that authenticates with same OIDC SSO identity provider and log in
- Navigate to ArgoCD app for the first time in the same browser session, but different tab
- **Here you will always be prompted to click on your IdP login icon (e.g., Keycloak or Okta)**
- However, you won't actually be required to re-login with IdP, because you're already authenticated in the browser, and will be taken inside ArgoCD a few seconds later

**Describe the bug**

I currently use Keycloak as an OIDC identity provider for ArgoCD and a number of others applications, such as Gitlab and Kubeflow.  For virtually all of my our applications if I complete the single sign on process once in a browser session, I am able to navigate to any other OIDC app in any other browser tab and am immediately logged into to that app without being prompted with a new login screen that asks me to re-authenticate.

For example, if I open a fresh incognito session in Chrome and log into my Gitlab account via Keycloak, I am then able to navigate to Kubeflow and have my email/identity affirmed in that app without being redirected to Keycloak again.  **However, when I have ArgoCD OIDC SSO enabled. I am always prompted by the ArgoCD login screen the first time I navigate to the app within a browser session -- regardless if I have already authenticated elsewhere.**

I think if would be great if the **auth mechanism that ArgoCD is using could automatically capture OIDC credentials from the current browser session and skip the login prompt which forces users to click on their identity provider's icon**.

**To Reproduce**

Pasted below is my copy of the config map I use for OIDC auth.   If one were to use this setup with their own IdP,  they would see an IdP authentication option at the ArgoCD `/login` screen. 

*argocd-cm.yaml*

apiVersion: v1
kind: ConfigMap
  name: argocd-cm
  namespace: argocd
data:
  url: https://argocd.XX-XX-XX
  users.anonymous.enabled: "false"
  oidc.config: |
    name: Keycloak
    issuer: https://keycloak.XX-XX-XX.io/realms/XX-XX-XX
    clientID: argocd-test-client
    clientSecret: XX-XX-XX
    requestedScopes: ["openid", "profile", "email", "groups"]
    requestedIDTokenClaims: {"groups": {"essential": true}, "email": {"essential": true}}

**Expected behavior**

If I am already authenticated in my browser I would like to seamlessly navigate to ArgoCD without being prompted with a login screen.  Especially if the login screen merely forces me to click on my IdP's name (e.g., Keycloak) but then immediately just takes me into the ArgoCD app.

**Version**

```shell
argocd: v1.3.0+c0e9315.dirty
  BuildDate: 2019-10-12T17:05:34Z
  GitCommit: c0e9315025f64710ec115904491c7fe95d7e822b
  GitTreeState: dirty
  GoVersion: go1.12.6
  Compiler: gc
  Platform: linux/amd64
```

**Logs**

Below are logs of the auth flow when I navigate to ArgoCd for the first time in a browser session that I have already authenticated via another app (e.g., Gitlab) and the same IdP.

```
time="2019-10-22T00:29:09Z" level=info msg="RequestedClaims: map[email:essential:true  groups:essential:true ]\n"
time="2019-10-22T00:29:09Z" level=info msg="Performing authorization_code flow login: https://keycloak.XX-XX-XX/realms/XX-XX-XX/protocol/openid-connect/auth?claims=%7B%22id_token%22%3A%7B%22email%22%3A%7B%22essential%22%3Atrue%7D%2C%22groups%22%3A%7B%22essential%22%3Atrue%7D%7D%7D&client_id=argocd-XX-XX-XX&redirect_uri=https%3A%2F%2Fapps.XX-XX-XX%2Fauth%2Fcallback&response_type=code&scope=openid+profile+email+groups&state=DwwfNmEQuh"
time="2019-10-22T00:29:09Z" level=info msg="received unary call /cluster.SettingsService/Get" grpc.method=Get grpc.request.claims=null grpc.request.content= grpc.service=cluster.SettingsService grpc.start_time="2019-10-22T00:29:09Z" span.kind=server system=grpc
time="2019-10-22T00:29:09Z" level=info msg="finished unary call with code OK" grpc.code=OK grpc.method=Get grpc.service=cluster.SettingsService grpc.start_time="2019-10-22T00:29:09Z" grpc.time_ms=1.479 span.kind=server system=grpc
time="2019-10-22T00:29:09Z" level=info msg="received unary call /cluster.SettingsService/Get" grpc.method=Get grpc.request.claims=null grpc.request.content= grpc.service=cluster.SettingsService grpc.start_time="2019-10-22T00:29:09Z" span.kind=server system=grpc
time="2019-10-22T00:29:09Z" level=info msg="finished unary call with code OK" grpc.code=OK grpc.method=Get grpc.service=cluster.SettingsService grpc.start_time="2019-10-22T00:29:09Z" grpc.time_ms=1 span.kind=server system=grpc
time="2019-10-22T00:29:09Z" level=info msg="received unary call /session.SessionService/GetUserInfo" grpc.method=GetUserInfo grpc.request.claims="{\"acr\":\"0\",\"aud\":\"argocd-aiops-platform-io\",\"auth_time\":1571704133,\"azp\":\"argocd-aiops-platform-io\",\"email\":\"bbhuston@XX-XX-XX.com\",\"email_verified\":true,\"exp\":1571704449,\"family_name\":\"user\",\"given_name\":\"test\",\"iat\":1571704149,\"iss\":\"https://keycloak.XX-XX-XX/realms/XX-XX-XX\",\"jti\":\"b0a2a507-c822-4a78-a4a1-be2000fe81c9\",\"name\":\"test user\",\"nbf\":0,\"preferred_username\":\"test-user\",\"session_state\":\"baf07987-193f-43dd-9720-292aa050fb9a\",\"sub\":\"9f766693-b373-4fc4-ad93-27419a81022d\",\"typ\":\"ID\"}" grpc.request.content= grpc.service=session.SessionService grpc.start_time="2019-10-22T00:29:09Z" span.kind=server system=grpc
time="2019-10-22T00:29:09Z" level=info msg="finished unary call with code OK" grpc.code=OK grpc.method=GetUserInfo grpc.service=session.SessionService grpc.start_time="2019-10-22T00:29:09Z" grpc.time_ms=1.115 span.kind=server system=grpc```

**Comment by alexec** [comment]
It sounds like you can work-around this by logging in?

**Comment by alexmt** [comment]
Fixed in 1.3.5
